src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -0,0 +1,196 @@
+        """
+        Runs the basic setup and initialization for the
+        MS Transferor module
+        :param microConfig: microservice configuration
+        """
+        super(MSMonitor, self).__init__(msConfig, logger)
+
+    def execute(self, reqStatus):
+        """
+        Executes the MS monitoring logic
+        :param reqStatus: request statue to process
+        :return:
+        """
+        try:
+            # get requests from ReqMgr2 data-service for given statue
+            # here with detail=False we get back list of records
+            requests = self.reqmgr2.getRequestByStatus([reqStatus], detail=False)
+            self.logger.debug('+++ monit found %s requests in %s state',
+                              len(requests), reqStatus)
+
+            requestStatuses = {}  # keep track of request statuses
+            for reqName in requests:
+                req = {'name': reqName, 'reqStatus': reqStatus}
+                self.logger.debug("+++ request %s", req)
+                # obtain status records from couchdb for given request
+                statusRecords = self.getStatusRecords(reqName)
+                # check status records and obtain completion status
+                statusRecords, completed = self.checkStatusRecords(statusRecords)
+                # if all transfers are completed
+                # move the request status staging -> staged
+                if completed == 100:
+                    self.logger.debug(
+                        "+++ request %s all transfers are completed", req)
+                    self.change(req, 'staged', '+++ monit')
+                # if pileup transfers are completed AND
+                # some input blocks are completed,
+                # move the request status staging -> staged
+                elif self.pileupTransfersCompleted(reqName):
+                    self.logger.debug(
+                        "+++ request %s pileup transfers are completed", req)
+                    self.change(req, 'staged', '+++ monit')
+                # transfers not completed
+                # just update the database with their completion
+                else:
+                    self.logger.debug(
+                        "+++ request %s transfers are %s completed",
+                        req, completed)
+                    # for us workflow name is the same as request name
+                    wname = reqName
+                    requestStatuses[wname] = statusRecords
+            self.updateTransferInfo(requestStatuses)
+        except Exception as err:  # general error
+            self.logger.exception('+++ monit error: %s', str(err))
+
+    def getStatusRecords(self, reqName):
+        """
+        Get transfer statuses for given request name
+        transfer docs on backend has the following form
+        https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+        So far logic is based on option D of the records
+
+        .. doctest::
```

Comment:
what 's this `.. doctest::` for?

Reply:
It is sphinx placeholder for code (cut-and-paste feature), see https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html?highlight=doctest#doctest-blocks